### PR TITLE
Adds book burning

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

For some reason, unlike other paper items, you couldn't ignite books with a lighter or other hot item.
Now, you can!

(Also made it so the odd obscure mechanic of cutting pages out of books can be done by any sharp item, not just wirecutters or knives)

## Why It's Good For The Game

Consistency. 
~~Burning copies of WGW.~~

## Changelog

:cl:
add: Books can now be burned just like any other paper item.
add: You can cut pages out of books with any sharp item, not just knives or wirecutters.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
What about Codex Cicatrix? Its not pathed like a book from what i remember

**Comment by ChungusGamer666** [comment]
> What about Codex Cicatrix? Its not pathed like a book from what i remember

yeah it ain't, but it ain't flammable either so it doesn't matter

**Comment by Wallemations** [comment]
I think it would be funny if the Bible (and all of its subtypes) had some sort of effect when you burned them, not anything too big but at least some form of visual effect?

**Comment by Helg2** [comment]
Does flame bible burns?

**Comment by ChungusGamer666** [comment]
> I think it would be funny if the Bible (and all of its subtypes) had some sort of effect when you burned them, not anything too big but at least some form of visual effect?

Hilarious suggestion, I have something in mind but it will have to be done in a separate PR after this one

**Comment by ChungusGamer666** [comment]
> Does flame bible burns?

It probably does, I don't think any bible is fireproof.
Honestly, it would be fitting if anything.

**Comment by OneAsianTortoise** [comment]
No way, fahrenheit 451

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -351,25 +351,28 @@
...
-/obj/item/proc/burn_paper_product_attackby_check(obj/item/I, mob/living/user, bypass_clumsy)
-	var/ignition_message = I.ignition_effect(src, user)
+/obj/item/proc/burn_paper_product_attackby_check(obj/item/attacking_item, mob/living/user, bypass_clumsy = FALSE)
+	//can't be put on fire!
+	if((resistance_flags & FIRE_PROOF) || !(resistance_flags & FLAMMABLE))
+		return
+	var/ignition_message = attacking_item.ignition_effect(src, user)
 	if(!ignition_message)
 		return
 	. = TRUE
```
Comment (san7890) [inline code/modules/paperwork/paper.dm:361]: this makes the whole proc really fucking silly so while we're here let's fix this in the best way possible

```suggestion
```

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -351,25 +351,28 @@
 		return TRUE
 	return ..()
 
-/obj/item/proc/burn_paper_product_attackby_check(obj/item/I, mob/living/user, bypass_clumsy)
-	var/ignition_message = I.ignition_effect(src, user)
+/obj/item/proc/burn_paper_product_attackby_check(obj/item/attacking_item, mob/living/user, bypass_clumsy = FALSE)
+	//can't be put on fire!
+	if((resistance_flags & FIRE_PROOF) || !(resistance_flags & FLAMMABLE))
+		return
```
Comment (san7890) [inline code/modules/paperwork/paper.dm:357]:
```suggestion
		return FALSE
```

File: code/modules/paperwork/paper.dm [hunk 3]
```diff
@@ -351,25 +351,28 @@
...
+	var/ignition_message = attacking_item.ignition_effect(src, user)
 	if(!ignition_message)
 		return
```
Comment (san7890) [inline code/modules/paperwork/paper.dm:360]:
```suggestion
		return FALSE
```

File: code/modules/paperwork/paper.dm [hunk 4]
```diff
@@ -351,25 +351,28 @@
...
 		user.visible_message(span_warning("[user] accidentally ignites [user.p_them()]self!"), \
 							span_userdanger("You miss [src] and accidentally light yourself on fire!"))
-		if(user.is_holding(I)) //checking if they're holding it in case TK is involved
-			user.dropItemToGround(I)
-		user.adjust_fire_stacks(1)
+		if(user.is_holding(attacking_item)) //checking if they're holding it in case TK is involved
+			user.dropItemToGround(attacking_item)
+		user.adjust_fire_stacks(attacking_item)
 		user.ignite_mob()
 		return
```
Comment (san7890) [inline code/modules/paperwork/paper.dm:369]:
```suggestion
		return TRUE
```

File: code/modules/paperwork/paper.dm [hunk 5]
```diff
@@ -351,25 +351,28 @@
...
 
 	if(user.is_holding(src)) //no TK shit here.
 		user.dropItemToGround(src)
 	user.visible_message(ignition_message)
 	add_fingerprint(user)
-	fire_act(I.get_temperature())
+	fire_act(attacking_item.get_temperature())
```
Comment (san7890) [inline code/modules/paperwork/paper.dm:375]:
```suggestion
	fire_act(attacking_item.get_temperature())
	return TRUE
```

**Review by san7890 — CHANGES_REQUESTED** [review]
it's reallllyyyy silly to have a null/TRUE dichotomy so let's at the very least patch out the dot operator and make a proper TRUE/FALSE boolean proc. if you are spooked about runtimes in this proc (lol) then just use `return .` in those spots

**Comment by DATA-xPUNGED** [comment]
THANK YOU

**Comment by ATH1909** [comment]
have you tested this with, say, hollow bibles?

**Comment by ChungusGamer666** [comment]
> have you tested this with, say, hollow bibles?

No i haven't

**Comment by Hardly3D** [comment]
An amazing addition. I'd love to see this added to paintings to deal with copies of awooga art.

**Pull request merged**